2.26.2: Test failures for `test_uv_resolver.py` ("uv mode doesn't support non-virtual environments")

### Describe the bug

Hi! 👋 

When upgrading the package on Arch Linux to 2.26.2 I ran into several test failures for the uv integration, with "uv mode doesn't support non-virtual environments" as error message:

```
=========================== short test summary info ============================
SKIPPED [1] tests/test_integration.py:47: Need at least 2 Python versions to test
SKIPPED [1] tests/test_integration.py:58: Need at least 2 Python versions to test
SKIPPED [1] tests/test_integration.py:110: Need at least 2 Python versions to test
SKIPPED [1] tests/test_integration.py:123: Need at least 2 Python versions to test
SKIPPED [1] tests/test_utils.py:171: Windows test
XFAIL tests/cli/test_others.py::test_search_package - HTTP search risk controlled
FAILED tests/cli/test_install.py::test_uv_install_pep582_not_allowed - RuntimeError: Call command ['add', 'requests', '--no-sync'] failed(1): See /tmp/pytest-of-builduser/pytest-0/test_uv_install_pep582_not_all0/logs/pdm-lock-7a3zjkqf.log for detailed deb
ug log.
[ProjectError]: uv mode doesn't support non-virtual environments
FAILED tests/resolver/test_uv_resolver.py::test_resolve_requirements - pdm.exceptions.ProjectError: uv mode doesn't support non-virtual environments
FAILED tests/resolver/test_uv_resolver.py::test_resolve_vcs_requirement - pdm.exceptions.ProjectError: uv mode doesn't support non-virtual environments
FAILED tests/resolver/test_uv_resolver.py::test_resolve_with_python_requires - pdm.exceptions.ProjectError: uv mode doesn't support non-virtual environments
FAILED tests/resolver/test_uv_resolver.py::test_resolve_dependencies_with_nested_extras - pdm.exceptions.ProjectError: uv mode doesn't support non-virtual environments
FAILED tests/resolver/test_uv_resolver.py::test_resolve_dependencies_with_overrides[2.31.0] - pdm.exceptions.ProjectError: uv mode doesn't support non-virtual environments
FAILED tests/resolver/test_uv_resolver.py::test_resolve_dependencies_with_overrides[==2.31.0] - pdm.exceptions.ProjectError: uv mode doesn't support non-virtual environments
= 7 failed, 1050 passed, 5 skipped, 3 deselected, 1 xfailed, 118 warnings in 56.83s =
```

Full build and check logs:

[python-pdm-2.26.2-1-x86_64-build.log](https://github.com/user-attachments/files/24051560/python-pdm-2.26.2-1-x86_64-build.log)

[python-pdm-2.26.2-1-x86_64-check.log](https://github.com/user-attachments/files/24051561/python-pdm-2.26.2-1-x86_64-check.log)


Should I disable these tests in our build environment?

### To reproduce

This is how we are building and testing for 2.26.1 (which is basically the same for 2.26.2): https://gitlab.archlinux.org/archlinux/packaging/packages/python-pdm/-/blob/8fd67f5e68196263929e0ac287436b328173ed21/PKGBUILD

### Expected Behavior

All tests pass.

### Environment Information

n/a

### Verbose Command Output

_No response_

### Additional Context

_No response_

### Are you willing to submit a PR to fix this bug?

- [ ] Yes, I would like to submit a PR.

--- a/news/3698.bugfix.md
+++ b/news/3698.bugfix.md
@@ -1,0 +1,1 @@
+Fix test failures with uv test cases using non-venv Python interpreters.

--- a/src/pdm/resolver/uv.py
+++ b/src/pdm/resolver/uv.py
@@ -9,7 +9,6 @@ from pathlib import Path
 from typing import TYPE_CHECKING, Any
 
 from pdm._types import HiddenText
-from pdm.exceptions import ProjectError
 from pdm.models.candidates import Candidate
 from pdm.models.markers import get_marker
 from pdm.models.repositories import Package
@@ -193,8 +192,12 @@ class UvResolver(Resolver):
         locked_repo = self.locked_repository or self.project.get_locked_repository()
         with uv_file_builder(self.project, str(self.target.requires_python), self.requirements, locked_repo) as builder:
             venv_project = self.environment.interpreter.get_venv()
-            if venv_project is None:
-                raise ProjectError("uv mode doesn't support non-virtual environments")
+            if venv_project is not None:
+                python_home = venv_project.root
+            else:  # pragma: no cover
+                python_home = self.environment.interpreter.path.parent
+                if python_home.name in ("bin", "Scripts"):
+                    python_home = python_home.parent
             builder.build_pyproject_toml()
             uv_lock_path = self.project.root / "uv.lock"
             if self.update_strategy != "all":
@@ -205,7 +208,7 @@ class UvResolver(Resolver):
                 uv_lock_command = self._build_lock_command()
                 self.project.core.ui.echo(f"Running uv lock command: {uv_lock_command}", verbosity=Verbosity.DETAIL)
                 real_command = [s.secret if isinstance(s, HiddenText) else s for s in uv_lock_command]
-                env = {**os.environ, "UV_PROJECT_ENVIRONMENT": str(venv_project.root)}
+                env = {**os.environ, "UV_PROJECT_ENVIRONMENT": str(python_home)}
                 subprocess.run(real_command, cwd=self.project.root, check=True, env=env)
             finally:
                 if isinstance(self.reporter, RichLockReporter):

